packages/erk-shared/src/erk_shared/context/context.py

```diff
@@ -0,0 +1,138 @@
+from erk_shared.github.issues import GitHubIssues
+from erk_shared.github.types import RepoInfo
+from erk_shared.integrations.completion import Completion
+from erk_shared.integrations.feedback import UserFeedback
+from erk_shared.integrations.graphite.abc import Graphite
+from erk_shared.integrations.shell import Shell
+from erk_shared.integrations.time.abc import Time
+from erk_shared.objectives.storage import ObjectiveStore
+from erk_shared.plan_store.store import PlanStore
+from erk_shared.project_discovery import ProjectContext
+from erk_shared.prompt_executor import PromptExecutor
+
+if TYPE_CHECKING:
+    # These ABCs stay in erk package due to their dependencies.
+    # Imported only for documentation - actual type hints use Any to avoid import cycles.
+    from erk.core.claude_executor import ClaudeExecutor as ClaudeExecutor
+    from erk.core.config_store import ConfigStore as ConfigStore
+    from erk.core.planner.registry_abc import PlannerRegistry as PlannerRegistry
+    from erk.core.script_writer import ScriptWriter as ScriptWriter
+    from erk.core.services.plan_list_service import PlanListService as PlanListService
+
+
+@dataclass(frozen=True)
+class ErkContext:
+    """Immutable context holding all dependencies for erk and dot-agent-kit operations.
+
+    Created at CLI entry point and threaded through the application via Click's
+    context system. Frozen to prevent accidental modification at runtime.
+
+    This unified context replaces both the old ErkContext (from erk.core.context)
+    and DotAgentContext (from dot_agent_kit.context).
+
+    Note:
+    - global_config may be None only during init command before config is created.
+      All other commands should have a valid GlobalConfig.
+    - Factory methods (minimal, for_test) are in erk.core.context, not here.
+
+    DotAgentContext Compatibility:
+    - github_issues -> issues (renamed for consistency)
+    - repo_root property -> repo.root (access via repo property or require_repo_root helper)
+    - debug field -> debug (preserved)
+    """
+
+    # Gateway integrations (from erk_shared)
+    git: Git
+    github: GitHub
+    issues: GitHubIssues  # Note: ErkContext naming (was github_issues in DotAgentContext)
+    graphite: Graphite
+    time: Time
+    session_store: ClaudeCodeSessionStore
+    plan_store: PlanStore
+    objectives: ObjectiveStore
+    prompt_executor: PromptExecutor  # From DotAgentContext
+
+    # Shell/CLI integrations (moved to erk_shared)
+    shell: Shell
+    completion: Completion
+    feedback: UserFeedback
+
+    # Erk-specific (ABCs stay in erk, injected as Any to avoid import cycle)
+    # Type hints are provided for static analysis via TYPE_CHECKING
+    claude_executor: Any  # ClaudeExecutor at runtime
+    config_store: Any  # ConfigStore at runtime
+    script_writer: Any  # ScriptWriter at runtime
+    planner_registry: Any  # PlannerRegistry at runtime
+    plan_list_service: Any  # PlanListService at runtime
```

Comment:
Move ABCs to erk-shared so we don't need to do this

Reply:
Acknowledged - Moving ABCs (ClaudeExecutor, ConfigStore, etc.) to erk-shared is a larger refactor that should be done in a follow-up PR. This PR establishes the context unification foundation.

_Addressed via `/erk:pr-address` at 2025-12-10 05:27 EST_